### PR TITLE
bsdlib: version 0.7.1

### PR DESCRIPTION
**Pull request by evenl**
Library version: 7dfb1ae80e78463f3b2affb5d096a25683949417

 * Fixed issue when FOTA would hang after reboot.
 * Added some more GNSS documentation.
 * Changing socket mode from non-blocking to blocking when there is a
   pending connect will now give an error.

Corresponding manifest update: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2202

Signed-off-by: Even Falch-Larsen <even.falch-larsen@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
@b-gent Please have a look at the changelog.

**Comment by b-gent** [comment]
I added some comments pointing out that the documentation in header files should not be there. It is simply not the right place for this information.

There is ongoing work on GNSS conceptual doc in PR #156 , I'm not sure if this information is included there. If it is, it can be removed from these headers. If it's not, it should be moved from the headers into RST.

These changes are not required for this release.

**Comment by lemrey** [comment]
@evenl we should probably move part of this content to the conceptual documentation like @b-gent suggested.

File: bsdlib/include/nrf_socket.h [hunk 1]
```diff
@@ -269,8 +269,50 @@ typedef uint32_t nrf_fd_set;
...
+ * Identifies the option used to set the GNSS fix interval in seconds.
+ *
+ * @details
+ * Fix interval parameter controls the time between GNSS receiver starts.
+ * It also controls whether the GNSS receiver is stopped after a valid PVT
+ * estimate. The fix interval parameter determines the mode of navigation.
+ * There are three different navigation modes available: single-fix,
+ * continuous, and periodic.
+ *
+ * Single-fix navigation mode is engaged by setting the fix interval to 0.
```
[the remark this answers is not in the document]
Comment (evenl) [reply bsdlib/include/nrf_socket.h:284]: That would mean that you will have to read the conceptual documentation to actually understand how to use this module. The API documentation should be the logical document to confer with if you want information about how to use one specific parameters.

File: bsdlib/include/nrf_socket.h [hunk 2]
```diff
@@ -956,6 +1017,44 @@ typedef uint16_t nrf_gnss_nmea_mask_t;
  *          - @c NRF_GNSS_PSM_DISABLED for no power mode policy.
  *          - @c NRF_GNSS_PSM_DUTY_CYCLING_PERFORMANCE for low power mode with better performance.
  *          - @c NRF_GNSS_PSM_DUTY_CYCLING_POWER for low power mode with lower power consumption.
+ *
+ * The GNSS receiver may track signals (i.e. run) either continuously or in duty
```
[the remark this answers is not in the document]
Comment (evenl) [reply bsdlib/include/nrf_socket.h:1021]: I would give the same argument. It's a bit strange to me that I have to confer with two documents to get all information I need for basic use of this module.

**Comment by b-gent** [comment]
Thanks a lot for the changes!

**Pull request merged**